### PR TITLE
feat: Add `HasWorldReference` mixin

### PR DESCRIPTION
**Pull request by spydon**
# Description

Adds a mixin for components similar to  `HasAncestor`, `HasParent` and `HasGameReference` but which provides access to the `World` which the component has as an ancestor.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: doc/flame/components.md [hunk 1]
```diff
@@ -189,6 +189,28 @@ that they will appear in the children list in the same order as they were
 scheduled for addition.
 
 
+### Access to the World from a Component
+
+When a component requires access to the `World` that it is attached to one can
```
Comment (erickzanardo) [inline doc/flame/components.md:194]: Unsure how to improve but the "that is attached to one" seems a bit confusing to me, maybe you meant when the user need to attach and have access to a world instance?

**Review by ufrshubham — COMMENTED** [review]
Not sure why a separate mixin is needed for this. Wouldn't `HasAncestor<MyWorld>` achieve the same result?

**Comment by spydon** [comment]
> Not sure why a separate mixin is needed for this. Wouldn't `HasAncestor<MyWorld>` achieve the same result?

It would, but the world will be very used in the upcoming release and when you use `HasAncestor` you probably want to use it for another ancestor than the root world a lot of the time and you can't use the mixin twice.

**Comment by ufrshubham** [comment]
Okay, in that case, should we call it `HasAncestorWorld` or `HasWorldAncestor`? Or maybe `BelongsToWorld`, `DescendantOfWorld`. `HasWorld` makes me think that the component is saying it has a world within it 😅

**Review by ufrshubham — APPROVED** [review]

**Comment by spydon** [comment]
> Okay, in that case, should we call it `HasAncestorWorld` or `HasWorldAncestor`? Or maybe `BelongsToWorld`, `DescendantOfWorld`. `HasWorld` makes me think that the component is saying it has a world within it 😅

Haha we actually discussed exactly that in the chat yesterday! It was previously called `HasWorldReference` to be in line with `HasGameReference`, Luan also had the same argument as you, so time to change back I think (or to `HasWorldAncestor` maybe).

**Review by wolfenrain — APPROVED** [review]

**Pull request merged**